app/services/db/private-cloud-costs.ts

```diff
@@ -0,0 +1,191 @@
+  });
+  return unitPrices;
+}
+
+export async function getCostItemsForRange(licencePlate: string, startDate: Date, endDate: Date): Promise<CostItem[]> {
+  const [unitPrices, allRequests] = await Promise.all([
+    getUnitsPrices(endDate),
+    prisma.privateCloudRequest.findMany({
+      where: {
+        licencePlate,
+        decisionStatus: { in: [DecisionStatus.PROVISIONED] },
+        provisionedDate: { not: null, lte: endDate },
+        OR: [{ type: RequestType.CREATE }, { isQuotaChanged: true }],
+      },
+      include: { decisionData: true },
+      orderBy: { provisionedDate: 'asc' },
+    }),
+  ]);
+
+  const changePoints = new Set<number>();
+
+  for (const price of unitPrices) {
+    changePoints.add(new Date(price.date).getTime());
+  }
+
+  for (const req of allRequests) {
+    if (req.provisionedDate) changePoints.add(new Date(req.provisionedDate).getTime());
+  }
+
+  changePoints.add(startDate.getTime());
+  changePoints.add(endDate.getTime());
+
+  const sortedPoints = _orderBy(Array.from(changePoints), undefined, 'asc').map((ms) => new Date(ms));
+
+  const costItems: CostItem[] = [];
+
+  for (let i = 0; i < sortedPoints.length - 1; i++) {
+    const intervalStart = sortedPoints[i];
+    const intervalEnd = sortedPoints[i + 1];
+
+    if (intervalEnd <= startDate || intervalStart >= endDate) continue;
+
+    const quota = _last(
+      _filter(
+        allRequests,
+        (
+          req,
+        ): req is (typeof allRequests)[number] & {
+          decisionData: { resourceRequests: Record<string, { cpu: number; storage: number }> };
+        } => !!req.provisionedDate && req.provisionedDate <= intervalStart,
+      ),
+    );
+
+    const price = _last(_filter(unitPrices, (p) => new Date(p.date) <= intervalStart)) ?? {
+      id: 'fallback-zero',
+      cpu: 0,
+      storage: 0,
+      date: intervalStart,
+    };
+
+    if (!quota || !price) continue;
+
+    let totalCpu = 0;
+    let totalStorage = 0;
+
+    if (!quota) continue;
+    const envs = quota.decisionData.resourceRequests;
+    for (const env of namespaceKeys) {
+      const usage = envs[env];
+      if (usage) {
+        totalCpu += usage.cpu || 0;
+        totalStorage += usage.storage || 0;
+      }
+    }
+
+    const durationMinutes = (intervalEnd.getTime() - intervalStart.getTime()) / (1000 * 60);
+    const durationFraction = durationMinutes / (1000 * 60 * 60 * 24);
+    const cpuCost = totalCpu * price.cpu * durationFraction;
+    const storageCost = totalStorage * price.storage * durationFraction;
+    const totalCost = cpuCost + storageCost;
+
+    costItems.push({
+      startDate: intervalStart,
+      endDate: intervalEnd,
+      cpu: totalCpu,
+      storage: totalStorage,
+      cpuCost,
+      storageCost,
+      totalCost,
+      unitPriceId: price.id,
+    });
+  }
+
+  return costItems;
+}
+
+export function estimateMonthTotal(currentTotal: number, year: number, month: number, lastItem?: CostItem) {
```

Comment:
let's add an item for the remaining days for the current month to display the item on the bottom table.